flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/abilities/source/SourceAbilitySpec.java

```diff
@@ -56,4 +57,12 @@
      */
     @JsonIgnore
     Optional<RowType> getProducedType();
+
+    /**
+     * When Filter,Limit,Partition,etc. are pushed into {@link DynamicTableSource}, the source will
```

Comment:
This description breaks the abstraction because it references specific implementations. How about this?

> Additional digests to generate when this spec is applied to the source.